streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -108,24 +92,63 @@ public void process(K key, Change<V> change) {
 
     }
 
-    private class KTableFilterValueGetter implements KTableValueGetter<K, V> {
+    @Override
+    public KTableValueGetterSupplier<K, V> view() {
+        // if the KTable is materialized, use the materialized store to return getter value;
+        // otherwise rely on the parent getter and apply filter on-the-fly
+        if (queryableName != null) {
+            return new KTableValueGetterSupplier<K, V>() {
+
+                public KTableValueGetter<K, V> get() {
+                    return new KTableMaterializedValueGetter<>(queryableName);
+                }
+
+                @Override
+                public String[] storeNames() {
+                    return new String[]{queryableName};
+                }
+            };
+        } else {
+            return new KTableValueGetterSupplier<K, V>() {
+                final KTableValueGetterSupplier<K, V> parentValueGetterSupplier = parent.valueGetterSupplier();
+
+                public KTableValueGetter<K, V> get() {
+                    return new KTableFilterValueGetter(parentValueGetterSupplier.get());
+                }
+
+                @Override
+                public String[] storeNames() {
+                    return parentValueGetterSupplier.storeNames();
+                }
+            };
+        }
+    }
 
+    private class KTableFilterValueGetter implements KTableValueGetter<K, V> {
         private final KTableValueGetter<K, V> parentGetter;
+        private KeyValueStore<K, V> store = null;
 
-        public KTableFilterValueGetter(KTableValueGetter<K, V> parentGetter) {
+        KTableFilterValueGetter(KTableValueGetter<K, V> parentGetter) {
             this.parentGetter = parentGetter;
         }
 
+        @SuppressWarnings("unchecked")
         @Override
         public void init(ProcessorContext context) {
-            parentGetter.init(context);
+            if (queryableName != null) {
+                store = (KeyValueStore<K, V>) context.getStateStore(queryableName);
+            } else {
+                parentGetter.init(context);
```

Comment:
Could be a gap in my understanding, but wouldn't the `parentGetter` already have been initialized?